src/vcf/record.cpp

```diff
@@ -344,33 +344,33 @@ namespace ebi
     void Record::strict_validation_info_predefined_tags(std::string const & field_key, std::string const & field_value,
                                                         std::vector<std::string> const & values) const
     {
-        if (field_key == "AA") {
+        if (field_key == AA) {
             static boost::regex aa_regex("((?![,;=])[[:print:]])+");
             if (!boost::regex_match(field_value, aa_regex)) {
                 throw new InfoBodyError{line, "INFO AA=" + field_value + " value is not a single dot or a string of bases", field_key};
             }
-        } else if (field_key == "AF") {
+        } else if (field_key == AF) {
             for (auto & value : values) {
                 if (std::stold(value) < 0 || std::stold(value) > 1) {
                     throw new InfoBodyError{line, "INFO AF=" + field_value + " value does not lie in the interval [0,1]", field_key};
                 }
             }
-        } else if (field_key == "CIGAR") {
+        } else if (field_key == CIGAR) {
             static boost::regex cigar_string("([0-9]+[MIDNSHPX])+");
             for (auto & value : values) {
                 if (!boost::regex_match(value, cigar_string)) {
                     throw new InfoBodyError{line, "INFO CIGAR=" + field_value + " value is not an alphanumeric string compliant with the SAM specification", field_key};
                 }
             }
-        } else if (field_key == "END") {
-            auto it = info.find("IMPRECISE");
+        } else if (field_key == END) {
+            auto it = info.find(IMPRECISE);
             if (it != info.end() && it->second == "0") {
                 auto expected = std::to_string(position + reference_allele.length() - 1);
```

Comment:
I think I had an oversight here, shouldn't this be done only if reference is not symbolic?